lib/puppet/util/windows/api_types.rb

```diff
@@ -269,6 +272,81 @@ class FILETIME < FFI::Struct
              :dwHighDateTime, :dword
     end
 
+    # https://docs.microsoft.com/en-us/previous-versions/windows/desktop/legacy/aa379560(v=vs.85)
+    # typedef struct _SECURITY_ATTRIBUTES {
+    #   DWORD  nLength;
+    #   LPVOID lpSecurityDescriptor;
+    #   BOOL   bInheritHandle;
+    # } SECURITY_ATTRIBUTES, *PSECURITY_ATTRIBUTES, *LPSECURITY_ATTRIBUTES;
+    class SECURITY_ATTRIBUTES < FFI::Struct
+      layout(
+        :nLength, :dword,
+        :lpSecurityDescriptor, :pointer,
+        :bInheritHandle, :int
```

Comment:
We've been using `:win32_bool` for `BOOL`, not to be confused with `BOOLEAN` which is `:uchar`